pkg/apiserver/profiling/router.go

```diff
@@ -303,7 +311,29 @@ func (s *Service) viewSingle(c *gin.Context) {
 		_ = c.Error(err)
 		return
 	}
-	c.Data(http.StatusOK, "image/svg+xml", content)
+
+	// set default content-type for legacy profiling content.
+	contentType := "image/svg+xml"
+
+	if task.ProfilingRawDataType == ProfilingRawDataTypeProtobuf {
+		switch outputType {
+		case string(ViewOutputTypeGraph):
+			svgContent, err := convertProtobufToSVG(content, task)
+			if err != nil {
+				_ = c.Error(err)
+				return
+			}
+			content = svgContent
+			contentType = "image/svg+xml"
+		case string(ViewOutputTypeFlameGraph):
```

Comment:
I totally agree with the relationships you mentioned above, but I don't think we have to change the name `ViewOutputTypeFlameGraph` to `ViewOutputTypeProtobuf`.

Since the OriginalSVG has been checked with the field `task.RawDataType`, it is empty for legacy record, so we can just return the content that been read from the legacy record file.

And for the new records, we should use the query `output_type` to find what kind of content the users want, if they want graph, we convert protobuf to svg, if they want flamegraph, we return the protobuf format. So I think the `ViewOutputType` should indicate users requirements not profiling raw data type.

Reply:
I may get your points. You mean we'd better to make the backend API more reasonable, since backend does not return a flamegraph not a protobuf content.

Reply:
> I may get your points. You mean we'd better to make the backend API more reasonable, since backend does not return a flamegraph not a protobuf content.

Yes. That's what I mean.